### DISCUSSION TITLE
How would i make the special stages from sonic 3 and knuckles for my game

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#80] [Q&A]
How would i make the special stages from sonic 3 and knuckles for my game

**Comment by DimensionWarped** [comment]
That's a broad question. Let's start with the following...

### The playfield - It is essentially just a big sprite (well, in reality it probably uses tile graphics, but for us in the year 2023 this is probably better achieved with sprites) with the following animations:

surge - move forward or backward one tile (backward is just the forward animation in reverse)
rotate left/right - turns the world 90 degrees (right is just the left animation played backwards)

You might need two copies of each animation on account of the whole checkerboard thing since every time you surge you are basically flipping the checkerboard. You could also just accomplish this with palette tricks though... which you'll likely need anyway if you want to have different colors per special stage.

### Checking for red orb loops (IE the only *actually* difficult thing about blue sphere)

This is a pretty complex geometry question honestly... it's going to involve a branching search of some kind starting from the orb that you just flipped. Basically, do a flood fill for each of the blue spheres adjacent to the sphere you just flipped, if you can make one contiguous shape with no gaps then see if you have a line of red spheres around the perimeter of that shape and if so, convert all of the blue spheres in the shape as well as the red spheres that form the perimeter of the loop to rings... and no, I can't immediately break this down more than that... it's something I'd have to sit down and struggle with probably for a whole day.

### Jumping

Pretty simple, essentially you just elevate your character sprite, turn it to a jumping animation, and after the first 10 or so frames and excluding the last 10 or so frames you disable collision with any map elements.



Now... I'm going to tell you why you shouldn't even be thinking about this right now... your focus should be on learning how to work within the engine to create things. This is essentially creating a whole new game and just grafting it onto the existing one. Really I shouldn't even be making commentary about how to accomplish this on this discussion board *except* for the fact that it has to do with a classic Sonic the Hedgehog game.

In any event, the whole point of a special stage back then was to show off technical tricks and if they aren't actually *your tricks*, then it's kind of pointless.

**Reply by Raptorq1** [reply]
Okay but I will eventually be make sonic 3 and knuckles in godot

**Reply by DimensionWarped** [reply]
That just sounds kind of pointless with both Sonic 3 Air and Sonic Origins already existing.

**Reply by Raptorq1** [reply]
Okay then how would you make the special stage from mania
